File: frontend/src/album.js
class Album {
    constructor(id, name, artist, genre, imgURL, songs, user){
        this.id = id
        this.name = name
        this.artist = artist
        this.genre = genre
        this.imgURL = imgURL
        this.songs = []
        songs.forEach(song => this.songs.push(song))
        this.user = user
        this.constructor.all.push(this)
    } 
    
    static all = []

}

File: beats-me-api/app/models/album.rb
class Album < ApplicationRecord
    has_many :songs
    belongs_to :artist
    belongs_to :genre
    # belongs_to :user

    def artist_name 
        self.artist ? self.artist.name : nil
    end

    def artist_name=(name)
        self.artist = Artist.find_or_create_by(name: name)
    end

    def genre_name=(name)
        self.genre = Genre.find_or_create_by(name: name)
    end

    def genre_name
        self.genre.name
    end

    def songs_attributes=(attributes_hash)
        attributes_hash.each do |k, value|
            song = Song.new(k)
            song.artist_id = self.artist_id
            song.album_id = self.id
            song.save
            self.songs << song unless song.title.blank?
        end
    end


end
